### ISSUE TITLE
com.twilio.exception.ApiConnectionException: Unable to deserialize response: No content to map due to end-of-input

### ISSUE THREAD

**Issue by Liam-Maloney** [twilio/twilio-java#398]
**Version: 7.4.0**

(Note the below logs relate to two distinct apps on two distinct app servers which use the same client library.)

Hi, 

We are using the Programmable Voice product.  Our production environment is an Ubuntu v14.04.3 instance hosted on Heroku.  Our application servers are running Play Framework v2.5.9, and we are using v7.4.0 of the Java Helper library (via Scala), with Version 1.4 of the Javascript SDK on the client side.

We periodically read our active queues from Twilio via: com.twilio.rest.api.v2010.account.Queue.reader(ACCOUNT_SID).read(), and perform redirection of queued calls to the correct Support Agents based on the result of this.  Our expected result is simply a list of all of our queues on Twilio, empty or not, which we can then filter on.  However, we are observing that this request is returning the following exception:

### Exception/Log
app1
```
[error] application - com.twilio.exception.ApiConnectionException: Unable to deserialize response: No content to map due to end-of-input
```
app2
```
[error] application - com.twilio.exception.ApiConnectionException: Unable to deserialize response: No content to map due to end-of-input 
app/web.1:  at [Source: ; line: 1, column: 0] 
app/web.1: JSON:  
```

### Steps to Reproduce
I am unsure of how to reproduce, as the problem seems to be intermittent with increasing frequency across two projects at the moment.  The following calls seem to trigger the client library to contact the Twilio rest api, but upon response, the client library encounters the deserialisation error attached above.

### Example Code

app1
```scala
Twilio.init(ACCOUNT_SID, AUTH_TOKEN)
val reader = com.twilio.rest.api.v2010.account.Queue.reader(ACCOUNT_SID)
reader.read().asScala.toList match { ...
```

app2
```scala
private val pollTwilioForCallQueues: (String, String, String) => UpdateCallQueues =
    (baseCallQueueName, accountSid, authToken) => {
      Twilio.init(accountSid, authToken)
      val (activeQueues, emptyQueues) = Queue.reader(accountSid).read().asScala
        .withFilter {_.getFriendlyName startsWith baseCallQueueName} map { queue =>
          CallQueueDigest(
              forPhoneNumber  = queue.getFriendlyName,
              numberOfCalls   = queue.getCurrentSize,
              averageWaitTime = queue.getAverageWaitTime,
              idOfQueue       = queue.getSid
            )
          } partition { _.numberOfCalls > 0 }
}
```

Any help would be greatly appreciated.

**Comment by efossier** [comment]
@Liam-Maloney Hi there, sorry for the late response, thank you for the thorough issue description. 
 As you mentioned, this seems to be a very rare and hard to debug issue, for what its worth this is the first time I've heard of this problem occurring. 

I'd like to suggest a couple of things you could do on your end to get a better idea of what exactly you are seeing:
- Does this happen across multiple resources or just the Call Queues?
- Can you instrument your code and perhaps log the request/response you are receiving from the API when this error occurs using the `getLastRequest`/`getLastResponse` methods on the HTTP Client (see https://github.com/twilio/twilio-java/blob/cd62281c69240c65a7081a4c40d03215cb7be49a/src/main/java/com/twilio/http/HttpClient.java#L15)? (This feature is available as of release `7.14.3`)

**Comment by thinkingserious** [comment]
Hello @Liam-Maloney,

Just following up here, are you still experiencing this issue? Have you tried @efossier's suggestions?

With best regards,

Elmer
